### PR TITLE
fix(Core): Adjustments to summoning ritual object handling.

### PR DESCRIPTION
**Pull request by heyitsbench**
## Changes Proposed:

This PR proposes changes to:
-  [X] Core (units, players, creatures, game systems).

## Issues Addressed:

- Progresses https://github.com/azerothcore/azerothcore-wotlk/issues/16101
- Progresses https://github.com/azerothcore/azerothcore-wotlk/issues/17012
- Closes https://github.com/chromiecraft/chromiecraft/issues/7218

## SOURCE:

The changes have been validated through:
- [X] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [X] Sniffs
- [X] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)

## Tests Performed:

This PR has been:
- [X] Tested in-game by the author.

## How to Test the Changes:

- [X] This pull request can be tested by following the reproduction steps provided in the linked issue

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/Entities/GameObject/GameObject.cpp [hunk 1]
```diff
@@ -1917,10 +1917,7 @@ void GameObject::Use(Unit* user)
                 if (level < info->meetingstone.minLevel)
                     return;
 
-                if (info->entry == 194097)
-                    spellId = 61994;                            // Ritual of Summoning
```
Comment (Kitzunu) [inline src/server/game/Entities/GameObject/GameObject.cpp:1921]: Why

Comment (heyitsbench) [reply src/server/game/Entities/GameObject/GameObject.cpp:1921]: 3.4.3 sniffs show no casting of 61994, but 23598 does show up.

Comment (Kitzunu) [reply src/server/game/Entities/GameObject/GameObject.cpp:1921]: Just to make sure you checked right. One is for summon stone one is for lock portal

Comment (heyitsbench) [reply src/server/game/Entities/GameObject/GameObject.cpp:1921]: Yes, I've been staring at sniffs for both of them for a couple days at this point, I am quite confident in what I've done in this PR (assuming the sniffs aren't missing information).

**Pull request merged**